fdbclient/NativeAPI.actor.cpp

```diff
@@ -3338,11 +3383,13 @@ ACTOR static Future<Void> tryCommit( Database cx, Reference<TransactionLogInfo>
 					commit_unknown_result()});
 		}
 
-		if (!req.tagSet.present()) {
-			wait(store(req.transaction.read_snapshot, readVersion));
+		if(req.tagSet.present() && tr->options.priority < TransactionPriority::IMMEDIATE){
+			wait(store(req.transaction.read_snapshot, readVersion) &&
+			     store(req.commitCostEstimation, estimateCommitCosts(tr, &req.transaction)));
```

Comment:
I think it's negligible because in `WriteTagThrottlingWorkload` which does read & write & random range clear, I compared the read/commit latency between two configurations (a. add tags to each transaction and b. no tags in each transaction), they are roughly same. Commit latency is around 1.2 ~ 2.3 ms. (The test is run on my local dev machine)